### DISCUSSION TITLE
Total kWh from CT1 and CT2 only?

### DISCUSSION THREAD

**Discussion by zwizard** [CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter#158] [Q&A] [answered]
Have clamps CT1 and CT2 connected to the mains coming into my breaker box. Is there a way to make the Total kWh just be from total of CT1 and CT2? Or have another sensor for the total of CT1 and CT2.

**Comment by pyrodex** [comment, marked as answer]
Are you running ESPHome? If so you could do the following:

This should be in your configuration already:
```
#Total Watts
  - platform: template
    name: ${disp_name} Total Watts
    id: totalWatts
    lambda: return id(ct1Watts).state + id(ct2Watts).state + id(ct3Watts).state + id(ct4Watts).state + id(ct5Watts).state + id(ct6Watts).state ;
    accuracy_decimals: 1
    unit_of_measurement: W
    device_class: energy
    update_interval: ${update_time}
```
Just copy and paste it and in the lambda line just remove the ones you don't care about in the SUM, also don't forget to rename it.

**Reply by zwizard** [reply]
I have this:
#Total Amps   
  - platform: template
    name: ${disp_name} Total Amps
    id: totalAmps
    lambda: return id(ct1Amps).state + id(ct2Amps).state + id(ct3Amps).state + id(ct4Amps).state + id(ct5Amps).state + id(ct6Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    device_class: current
    update_interval: ${update_time}         
#Total Watts
  - platform: template
    name: ${disp_name} Total Watts
    id: totalWatts
    lambda: return id(totalAmps).state * id(ic1Volts).state;
    accuracy_decimals: 1
    unit_of_measurement: W
    device_class: power
    update_interval: ${update_time}


Upon further review of the code, it looks like I just need to change:
lambda: return id(ct1Amps).state + id(ct2Amps).state + id(ct3Amps).state + id(ct4Amps).state + id(ct5Amps).state + id(ct6Amps).state ;
to
lambda: return id(ct1Amps).state + id(ct2Amps).state;

**Reply by CircuitSetup** [reply]
You should use the 'power' output from the meter instead of `id(totalAmps).state * id(ic1Volts).state;`. This leaves power factor out of the calculation and will return a higher power output than what you're actually using.

[Start with this example](https://github.com/CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter/blob/master/Software/ESPHome/6chan_energy_meter_main_board.yaml)

If all you want is total amps output from CT1, CT2, then `lambda: return id(ct1Amps).state + id(ct2Amps).state;` is correct, but calculating kWh for [total_daily_energy](https://github.com/CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter/blob/e570c130106fa9f0b199fc34e18aa7010dcea2a8/Software/ESPHome/6chan_energy_meter_main_board.yaml#L168) needs a [power input (Watts)](https://github.com/CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter/blob/e570c130106fa9f0b199fc34e18aa7010dcea2a8/Software/ESPHome/6chan_energy_meter_main_board.yaml#L159)

**Reply by zwizard** [reply]
Like this?
#Total Amps
  - platform: template
    name: ${disp_name} Total Amps
    id: totalAmps
    lambda: return id(ct1Amps).state + id(ct2Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    device_class: current
    update_interval: ${update_time}
    
#Total Watts
  - platform: template
    name: ${disp_name} Total Watts
    id: totalWatts
    lambda: return id(ct1Watts).state + id(ct2Watts).state ;
    accuracy_decimals: 1
    unit_of_measurement: W
    device_class: power
    update_interval: ${update_time}
    
#kWh
  - platform: total_daily_energy
    name: ${disp_name} Total kWh
    power_id: totalWatts
    filters:
      - multiply: 0.001
    unit_of_measurement: kWh
    device_class: energy
    state_class: total_increasing
